.github/workflows/build.yml

```diff
@@ -159,6 +159,19 @@ jobs:
         run: |
           script/check-posix-build
 
+  posix-spi:
+    name: posix-spi
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Bootstrap
+      run: |
+        sudo apt-get update
+        sudo apt-get --no-install-recommends install -y cmake ninja-build
+    - name: Build
+      run: |
+        mkdir build && cd build && cmake -GNinja -DOT_COMPILE_WARNING_AS_ERROR=on -DOT_PLATFORM=posix -DOT_POSIX_CONFIG_RCP_BUS=SPI .. && ninja
```

Comment:
Can we create `script/check-posix-build-cmake` and use that here?

Reply:
Updated. Also fixed another error ([scan format type mismatch](https://github.com/openthread/openthread/pull/4913/files#diff-19d73e39e7f769ebb868e2472105e02fR438)) found by CMake build.